daml-foundations/daml-ghc/daml-prim-src/GHC/Num.daml

```diff
@@ -108,10 +118,36 @@ instance Additive Decimal where
 instance Multiplicative Decimal where
     (*) = primitive @"BEMulDecimal"
     munit = 1.0
+    x ^ n
+        | n == 0 = 1.0
+        | n == 2 = x * x
+        | n < 0 = 1.0 / x ^ (negate n)
```

Comment:
I'm not convinced we want that. We should instead fail when the exponent is negative. This would also allow for moving the `Fractional` type class to where it came from. Any opinions @neil-da?